lib/active_model/serializer/adapter/attributes.rb

```diff
@@ -2,42 +2,58 @@ module ActiveModel
   class Serializer
     class Adapter
       class Attributes < Adapter
+        attr_accessor :_nested_associations
```

Comment:
Let's look at how emberdata and other libs do it.

B mobile phone

> On Sep 20, 2015, at 12:12 PM, L. Preston Sego III <notifications@github.com> wrote:
> 
> In lib/active_model/serializer/adapter/attributes.rb:
> 
> > @@ -2,42 +2,58 @@ module ActiveModel
> >    class Serializer
> >      class Adapter
> >        class Attributes < Adapter
> > +        attr_accessor :_nested_associations
> I think we need to extract the commonest functionality from all the adapters:
> 
> finding associations
> filtering associations
> iterating over those associations
> and iterating over nested associations.
> this can be done with yields, which should really help separate responsibility between adapters and their superclass, and also remove any hint of duplication.
> 
> —
> Reply to this email directly or view it on GitHub.
>